### PR TITLE
HEEDLS-805 synchronise groups bugfix

### PR DESCRIPTION
**Pull request by livzorn**
### JIRA link
_HEEDLS-805_

### Description
Fixed 2 issues in the group synchronisation process:
- Account for multiple groups created from the same field
- Use case insensitivity when looking for matching groups for adding/removing delegates

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

**Review by GeorgeBlackwell — APPROVED** [review]
Nice work 🚀

**Review by DanBloxham-sw — CHANGES_REQUESTED** [review]
Some very minor points/questions.

File: DigitalLearningSolutions.Data/Services/GroupsService.cs [hunk 1]
```diff
@@ -556,9 +556,13 @@ private IEnumerable<Group> GetSynchronisedGroupsForCentre(int centreId)
                 .Where(g => g.ChangesToRegistrationDetailsShouldChangeGroupMembership);
         }
 
-        private bool GroupLabelMatchesAnswer(string groupLabel, string? answer, string? linkedFieldName)
+        private static bool GroupLabelMatchesAnswer(string groupLabel, string? answer, string? linkedFieldName)
         {
-            return groupLabel == answer || groupLabel == linkedFieldName + " - " + answer;
+            return string.Equals(groupLabel, answer, StringComparison.CurrentCultureIgnoreCase) || string.Equals(
+                groupLabel,
+                linkedFieldName + " - " + answer,
```
[the remark this answers is not in the document]
Comment (livzorn) [reply DigitalLearningSolutions.Data/Services/GroupsService.cs]: Good idea, I've added a private method in this file to do that

File: DigitalLearningSolutions.Data/Helpers/LinkedFieldHelper.cs [hunk 2]
```diff
@@ -49,7 +49,7 @@ ICentreRegistrationPromptsService centreRegistrationPromptsService
             {
                 var oldJobGroup = jobGroupsDataService.GetJobGroupName(oldAnswers.JobGroupId);
                 var newJobGroup = jobGroupsDataService.GetJobGroupName(newAnswers.JobGroupId);
-                changedLinkedFieldsWithAnswers.Add(new LinkedFieldChange(RegistrationField.JobGroup.LinkedToFieldId, "Job Group", oldJobGroup, newJobGroup));
+                changedLinkedFieldsWithAnswers.Add(new LinkedFieldChange(RegistrationField.JobGroup.LinkedToFieldId, "Job group", oldJobGroup, newJobGroup));
```
[the remark this answers is not in the document]
Comment (livzorn) [reply DigitalLearningSolutions.Data/Helpers/LinkedFieldHelper.cs:52]: Oh yeah, I first made this change to fix another bug - when generating groups from job groups, the prefix added is "Job group" (to be consistent with the casing we use when displaying 'Job group' around the site), but here it checks for groups prefixed with "Job Group", so job groups with a prefix weren't being synchronised properly. Now that we've decided to treat the names case insensitively it doesn't make any difference at all, but it doesn't hurt to use the consistent casing.

File: DigitalLearningSolutions.Data.Tests/Services/GroupServiceTests/GroupsServiceSynchroniseGroupsTests.cs [hunk 3]
```diff
@@ -265,14 +272,124 @@ public void
...
+                    )
+                ).MustHaveHappenedOnceExactly();
+                A.CallTo(
+                    () => groupsDataService.RemoveRelatedProgressRecordsForGroup(
+                        groupId,
+                        reusableDelegateDetails.Id,
+                        false,
+                        testDate
+                    )
+                ).MustHaveHappened();
```
[the remark this answers is not in the document]
Comment (livzorn) [reply DigitalLearningSolutions.Data.Tests/Services/GroupServiceTests/GroupsServiceSynchroniseGroupsTests.cs]: yep, done :)

**Pull request merged**